src/textures/Texture.js

```diff
@@ -17,77 +17,81 @@ import { ImageUtils } from '../extras/ImageUtils.js';
 
 let textureId = 0;
 
-function Texture( image = Texture.DEFAULT_IMAGE, mapping = Texture.DEFAULT_MAPPING, wrapS = ClampToEdgeWrapping, wrapT = ClampToEdgeWrapping, magFilter = LinearFilter, minFilter = LinearMipmapLinearFilter, format = RGBAFormat, type = UnsignedByteType, anisotropy = 1, encoding = LinearEncoding ) {
+class Texture extends EventDispatcher {
 
-	Object.defineProperty( this, 'id', { value: textureId ++ } );
+	constructor( image = Texture.DEFAULT_IMAGE, mapping = Texture.DEFAULT_MAPPING, wrapS = ClampToEdgeWrapping, wrapT = ClampToEdgeWrapping, magFilter = LinearFilter, minFilter = LinearMipmapLinearFilter, format = RGBAFormat, type = UnsignedByteType, anisotropy = 1, encoding = LinearEncoding ) {
 
-	this.uuid = MathUtils.generateUUID();
+		super();
 
-	this.name = '';
+		Object.defineProperty( this, 'id', { value: textureId ++ } );
+		Object.defineProperty( this, 'isTexture', { value: true } );
+		Object.defineProperty( this, 'DEFAULT_IMAGE', { value: undefined } );
+		Object.defineProperty( this, 'DEFAULT_MAPPING', { value: UVMapping } );
```

Comment:
These should be static.

Reply:
@marcofugaro hey I run into an eslint warning trying to move these static props inside the class. They're currently where they originally where. I thought we could use class static props now because of #20395.

I tried:
```javascript
class Texture extends EventDispatcher {

	static DEFAULT_IMAGE = undefined;
	static DEFAULT_MAPPING = UVMapping;

	constructor( image
```
but eslint came up with:
```
C:\Users\rekke\Documents\three.js\src\textures\Texture.js
  22:23  error  Parsing error: Unexpected token =        

✖ 1 problem (1 error, 0 warnings)
```

Reply:
@DefinitelyMaybe ops, I didn't test with eslint, it also needs to be updated to support class properties.

Done it in #21236

Reply:
cool! looking forward to it :)

Reply:
The linter issue in context of `static` should be gone now. So the PR can be finished.